extension/searching.js

```diff
@@ -12,6 +12,49 @@ this.searching = (function() {
     return searchUrl.href;
   };
 
+  exports.ddgEntitySearch = async function(query) {
+    const response = await fetch(
+      `https://api.duckduckgo.com/?q=${query}&format=json&pretty=1&skip_disambig=1`
+    );
+    const ddgData = await response.json();
+    if (!ddgData.AbstractText) {
+      // the response from DDG was null, and there was no matching Instant Answer result
+      return {};
+    }
+    const cardData = (({
```

Comment:
I don't understand what this is doing.

Reply:
https://stackoverflow.com/questions/17781472/how-to-get-a-subset-of-a-javascript-objects-properties

The JSON that we get from DDG's API can be huge, with lots of keys we don't actually need for the card, so I wanted to just pick out the ones that actually get used within the UI. If that's unnecessary, I'm happy to cut this and pass along the full json? Or can add a comment on what it's doing.

Reply:
That's fine, but yeah, add a comment to clarify

Reply:
Done